{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d7bf8637",
   "metadata": {},
   "source": [
    "# **Multilocus Sequence Typing (MLST)**\n",
    "\n",
    "- **Analysis type:** MLST\n",
    "- **Input data:** Assembled genomes (FASTA)\n",
    "- **Organism:** *Acinetobacter baumannii*\n",
    "- **Typing scheme:** Oxford MLST\n",
    "\n",
    "Following genome assembly, **multilocus sequence typing (MLST)** was performed to assign sequence types (STs) to each isolate. MLST assigns isolates to sequence types based on the allelic combinations of housekeeping genes defined in a species-specific typing scheme. This allows comparison of isolates across studies and supports population structure analysis.\n",
    "\n",
    "For *Acinetobacter baumannii*, the Oxford MLST scheme was used.\n",
    "\n",
    "MLST was performed using **mlst (2.23.0)** (Torsten Seemann), a command-line tool that assigns sequence types by querying assembled genome contigs against curated PubMLST schemes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "24def3bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mlst 2.23.0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# initialise conda\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate mlst\n",
    "\n",
    "# check mlst\n",
    "mlst --version\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee057bba",
   "metadata": {},
   "source": [
    "## Input Files\n",
    "\n",
    "The MLST analysis was performed using assembled genome contigs generated from the GHRU assembly pipeline.\n",
    "\n",
    "### Input requirements\n",
    "\n",
    "- Assembled genomes in FASTA format\n",
    "- One assembly per sample\n",
    "- Assemblies must be free of contamination and of sufficient quality\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ddc51cd",
   "metadata": {},
   "source": [
    "## MLST Execution\n",
    "\n",
    "The analysis was performed in a Conda-managed environment, standard output was written to a results file and execution logs were retained for reproducibility."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0f8c8042",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# initialise conda\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate mlst\n",
    "\n",
    "# define paths\n",
    "ASSEMBLY_DIR=/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies\n",
    "MLST_OUTDIR=/data/internship_data/nidhi/aba/new_output/mlst_output\n",
    "LOGDIR=/data/internship_data/nidhi/aba/new_output/logs\n",
    "\n",
    "# create required directories\n",
    "mkdir -p $MLST_OUTDIR $LOGDIR\n",
    "\n",
    "# run MLST with logging\n",
    "mlst $ASSEMBLY_DIR/*.short.fasta \\\n",
    "> $MLST_OUTDIR/mlst_results.tsv \\\n",
    "2> $LOGDIR/mlst.log"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14a7bc02",
   "metadata": {},
   "source": [
    "## MLST Output and Downstream Processing\n",
    "\n",
    "The MLST tool generates a primary output containing sequence type (ST) and allelic profiles for each isolate. All additional files, including ST distributions, allele variation summaries, International Clone (IC) assignments, and visualizations, were generated through downstream data processing and analysis of the MLST results.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "79ac664e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies/ABA-1000.short.fasta  abaumannii_2  10    Pas_cpn60(1)  Pas_fusA(3)  Pas_gltA(2)   Pas_pyrG(1)  Pas_recA(4)  Pas_rplB(4)  Pas_rpoB(4)\n",
      "/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies/ABA-1001.short.fasta  abaumannii_2  2     Pas_cpn60(2)  Pas_fusA(2)  Pas_gltA(2)   Pas_pyrG(2)  Pas_recA(2)  Pas_rplB(2)  Pas_rpoB(2)\n",
      "/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies/ABA-1002.short.fasta  abaumannii_2  2     Pas_cpn60(2)  Pas_fusA(2)  Pas_gltA(2)   Pas_pyrG(2)  Pas_recA(2)  Pas_rplB(2)  Pas_rpoB(2)\n",
      "/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies/ABA-1003.short.fasta  abaumannii_2  2     Pas_cpn60(2)  Pas_fusA(2)  Pas_gltA(2)   Pas_pyrG(2)  Pas_recA(2)  Pas_rplB(2)  Pas_rpoB(2)\n",
      "/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies/ABA-1004.short.fasta  abaumannii_2  2     Pas_cpn60(2)  Pas_fusA(2)  Pas_gltA(2)   Pas_pyrG(2)  Pas_recA(2)  Pas_rplB(2)  Pas_rpoB(2)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "column -t /data/internship_data/nidhi/aba/new_output/mlst_output/mlst_results.tsv | head -n 5"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "aba_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
