{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a6ef5192",
   "metadata": {},
   "source": [
    "# **Antimicrobial Resistance Gene Identification – AMRFinderPlus**\n",
    "\n",
    "## Tool Information\n",
    "\n",
    "- **Tool:** AMRFinderPlus (4.0.15)\n",
    "- **Input:** Assembled genome contigs (FASTA)\n",
    "- **Organism:** *Acinetobacter baumannii*\n",
    "- **Analysis type:** Antimicrobial resistance gene detection\n",
    "- **Database:** NCBI AMRFinder database\n",
    "\n",
    "This notebook documents the identification of **antimicrobial resistance (AMR) genes** in *Acinetobacter baumannii* genomes using **AMRFinderPlus**.\n",
    "\n",
    "AMRFinderPlus is a tool developed by NCBI for the detection of antimicrobial resistance genes, resistance-associated point mutations, and stress response genes in bacterial genomes. It uses curated reference databases and hidden Markov models to identify resistance determinants with high accuracy.\n",
    "\n",
    "The analysis was performed on assembled genome contigs generated from the GHRU assembly pipeline. Identified AMR determinants are used for downstream resistance profiling and integration with mobile genetic element analyses. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "048fbb89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4.0.15\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# initialise conda\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate amrfinder\n",
    "\n",
    "# check installation\n",
    "amrfinder --version\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cf1d273",
   "metadata": {},
   "source": [
    "## Input Files\n",
    "\n",
    "The input for AMRFinderPlus consists of assembled genome contigs generated from the GHRU assembly pipeline.\n",
    "\n",
    "### Input requirements\n",
    "\n",
    "- Genome assemblies in FASTA format\n",
    "- One assembly per sample\n",
    "- Assemblies should be quality-checked prior to AMR analysis\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "570b31e6",
   "metadata": {},
   "source": [
    "## AMRFinder Database Status\n",
    "\n",
    "AMRFinderPlus uses a curated resistance gene database provided by NCBI. If the database is already present in the execution environment, it is reused automatically. Database initialization is only required once per environment.\n",
    "\n",
    "⚠️ Run the below code only if using AMRFinderPlus for the first time\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be5a2369",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate amrfinder\n",
    "\n",
    "amrfinder --update"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a25c861",
   "metadata": {},
   "source": [
    "The AMRFinderPlus resistance gene database had been generated during a prior setup step and was already available in the execution environment at the time of this analysis.\n",
    "As a result, database initialization was not repeated and the existing database was reused for all AMRFinderPlus runs."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3109a281",
   "metadata": {},
   "source": [
    "## AMRFinderPlus Execution\n",
    "\n",
    "AMRFinderPlus was executed on assembled genome contigs to identify antimicrobial\n",
    "resistance genes and resistance-associated features. Each genome assembly was\n",
    "processed individually using a loop-based execution strategy.\n",
    "\n",
    "The analysis was performed within a Conda-managed environment. Runtime messages\n",
    "and warnings were redirected to a dedicated log file to prevent verbose output\n",
    "in the Jupyter notebook while preserving full execution traceability."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d828a48e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate amrfinder\n",
    "\n",
    "ASSEMBLY_DIR=/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies\n",
    "AMR_OUTDIR=/data/internship_data/nidhi/aba/new_output/amrfinder_output\n",
    "LOGDIR=/data/internship_data/nidhi/aba/new_output/logs\n",
    "\n",
    "mkdir -p $AMR_OUTDIR $LOGDIR\n",
    "\n",
    "for fasta in $ASSEMBLY_DIR/*.short.fasta; do\n",
    "    sample=$(basename \"$fasta\" .short.fasta)\n",
    "\n",
    "    amrfinder \\\n",
    "        --nucleotide \"$fasta\" \\\n",
    "        --organism Acinetobacter_baumannii \\\n",
    "        --output \"$AMR_OUTDIR/${sample}_amrfinder.tsv\" \\\n",
    "        >> \"$LOGDIR/amrfinder.log\" 2>&1\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd0bb5ea",
   "metadata": {},
   "source": [
    "## Expected Outputs\n",
    "\n",
    "AMRFinderPlus generates per-sample tabular output files containing:\n",
    "\n",
    "- Identified AMR genes\n",
    "- Resistance classes and subclasses\n",
    "- Sequence identity and coverage\n",
    "- Evidence type (protein homolog, HMM, point mutation)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e566eb50",
   "metadata": {},
   "source": [
    "Each output file corresponds to one genome assembly and is used for downstream AMR profiling and integration with mobile genetic element analyses."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "79f8ecbe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein id\tContig id\tStart\tStop\tStrand\tElement symbol\tElement name\n",
      "NA\tcontig00001\t63453\t64601\t+\tblaADC-76\textended-spectrum class C beta-lactamase ADC-76\n",
      "NA\tcontig00008\t79325\t82036\t-\tgyrA_S81L\tAcinetobacter baumannii quinolone resistant GyrA\n",
      "NA\tcontig00024\t8554\t9339\t-\tant(3'')-IIa\taminoglycoside nucleotidyltransferase ANT(3'')-IIa\n",
      "NA\tcontig00039\t6418\t7239\t-\tblaOXA-68\tOXA-51 family carbapenem-hydrolyzing class D beta-lactamase OXA-68\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cut -f1-7 /data/internship_data/nidhi/aba/new_output/amrfinder_output/ABA-1000_amrfinder.tsv | head -n 5"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "aba_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
