{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a6ef5192",
   "metadata": {},
   "source": [
    "# **Antimicrobial Resistance Gene Identification – AMRFinderPlus**\n",
    "\n",
    "## Tool Information\n",
    "\n",
    "- **Tool:** AMRFinderPlus (4.0.15)\n",
    "- **Input:** Assembled genome contigs (FASTA)\n",
    "- **Organism:** *Acinetobacter baumannii*\n",
    "- **Analysis type:** Antimicrobial resistance gene detection\n",
    "- **Database:** NCBI AMRFinder database\n",
    "\n",
    "This notebook documents the identification of **antimicrobial resistance (AMR) genes** in *Acinetobacter baumannii* genomes using **AMRFinderPlus**.\n",
    "\n",
    "AMRFinderPlus is a tool developed by NCBI for the detection of antimicrobial resistance genes, resistance-associated point mutations, and stress response genes in bacterial genomes. It uses curated reference databases and hidden Markov models to identify resistance determinants with high accuracy.\n",
    "\n",
    "The analysis was performed on assembled genome contigs generated from the GHRU assembly pipeline. Identified AMR determinants are used for downstream resistance profiling and integration with mobile genetic element analyses. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "048fbb89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4.0.15\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# initialise conda\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate amrfinder\n",
    "\n",
    "# check installation\n",
    "amrfinder --version\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cf1d273",
   "metadata": {},
   "source": [
    "## Input Files\n",
    "\n",
    "The input for AMRFinderPlus consists of assembled genome contigs generated from the GHRU assembly pipeline.\n",
    "\n",
    "### Input requirements\n",
    "\n",
    "- Genome assemblies in FASTA format\n",
    "- One assembly per sample\n",
    "- Assemblies should be quality-checked prior to AMR analysis\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "570b31e6",
   "metadata": {},
   "source": [
    "## AMRFinder Database Status\n",
    "\n",
    "AMRFinderPlus uses a curated resistance gene database provided by NCBI. If the database is already present in the execution environment, it is reused automatically. Database initialization is only required once per environment.\n",
    "\n",
    "⚠️ Run the below code only if using AMRFinderPlus for the first time\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be5a2369",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate amrfinder\n",
    "\n",
    "amrfinder --update"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a25c861",
   "metadata": {},
   "source": [
    "The AMRFinderPlus resistance gene database had been generated during a prior setup step and was already available in the execution environment at the time of this analysis.\n",
    "As a result, database initialization was not repeated and the existing database was reused for all AMRFinderPlus runs."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3109a281",
   "metadata": {},
   "source": [
    "## AMRFinderPlus Execution\n",
    "\n",
    "AMRFinderPlus was executed on assembled genome contigs to identify antimicrobial\n",
    "resistance genes and resistance-associated features. Each genome assembly was\n",
    "processed individually using a loop-based execution strategy.\n",
    "\n",
    "The analysis was performed within a Conda-managed environment. Runtime messages\n",
    "and warnings were redirected to a dedicated log file to prevent verbose output\n",
    "in the Jupyter notebook while preserving full execution traceability."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d828a48e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "source /home/anaconda/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate amrfinder\n",
    "\n",
    "ASSEMBLY_DIR=/data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies\n",
    "AMR_OUTDIR=/data/internship_data/nidhi/aba/new_output/amrfinder_output\n",
    "LOGDIR=/data/internship_data/nidhi/aba/new_output/logs\n",
    "\n",
    "mkdir -p $AMR_OUTDIR $LOGDIR\n",
    "\n",
    "for fasta in $ASSEMBLY_DIR/*.short.fasta; do\n",
    "    sample=$(basename \"$fasta\" .short.fasta)\n",
    "\n",
    "    amrfinder \\\n",
    "        --nucleotide \"$fasta\" \\\n",
    "        --organism Acinetobacter_baumannii \\\n",
    "        --output \"$AMR_OUTDIR/${sample}_amrfinder.tsv\" \\\n",
    "        >> \"$LOGDIR/amrfinder.log\" 2>&1\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd0bb5ea",
   "metadata": {},
   "source": [
    "## Expected Outputs\n",
    "\n",
    "AMRFinderPlus generates per-sample tabular output files containing:\n",
    "\n",
    "- Identified AMR genes\n",
    "- Resistance classes and subclasses\n",
    "- Sequence identity and coverage\n",
    "- Evidence type (protein homolog, HMM, point mutation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "eb02dc68",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein  id           Contig  id     Start  Stop          Strand                 Element                 symbol                                  Element                name            Scope           Type            Subtype                     Class         Subclass                 Method                                   Target      length      Reference  sequence        length            %                       Coverage           of                                      reference              %               Identity                to              reference   Alignment       length    Closest      reference  accession    Closest  reference  name  HMM  accession  HMM  description\n",
      "NA       contig00001  63453   64601  +      blaADC-76     extended-spectrum      class                   C                                       beta-lactamase         ADC-76          core            AMR             AMR                         BETA-LACTAM   CEPHALOSPORIN            ALLELEX                                  383         383         100.00     100.00          383               WP_001211237.1          extended-spectrum  class                                   C                      beta-lactamase  ADC-76                  NA              NA                                                                                                                        \n",
      "NA       contig00008  79325   82036  -      gyrA_S81L     Acinetobacter          baumannii               quinolone                               resistant              GyrA            core            AMR             POINT                       QUINOLONE     QUINOLONE                POINTX                                   904         904         100.00     99.67           904               WP_000116450.1          DNA                gyrase                                  subunit                A               GyrA                    NA              NA                                                                                                                        \n",
      "NA       contig00024  8554    9339   -      ant(3'')-IIa  aminoglycoside         nucleotidyltransferase  ANT(3'')-IIa                            core                   AMR             AMR             AMINOGLYCOSIDE  SPECTINOMYCIN/STREPTOMYCIN  BLASTX        262                      262                                      100.00      98.85       262        WP_001279062.1  aminoglycoside    nucleotidyltransferase  ANT(3'')-IIa       NA                                      NA                                                                                                                                                                                                       \n",
      "NA       contig00039  6418    7239   -      blaOXA-68     OXA-51                 family                  carbapenem-hydrolyzing                  class                  D               beta-lactamase  OXA-68          core                        AMR           AMR                      BETA-LACTAM                              CARBAPENEM  ALLELEX     274        274             100.00            100.00                  274                WP_001021775.1                          OXA-51                 family          carbapenem-hydrolyzing  class           D           beta-lactamase  OXA-68    NA           NA                                                                     \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "column -t /data/internship_data/nidhi/aba/new_output/amrfinder_output/ABA-1000_amrfinder.tsv | head -n 5"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e566eb50",
   "metadata": {},
   "source": [
    "Each output file corresponds to one genome assembly and is used for downstream AMR profiling and integration with mobile genetic element analyses."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "aba_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
