{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "532393dc",
   "metadata": {},
   "source": [
    "# **Plasmid Replicon Typing  â€“ AcinetobacterPlasmidTyping**\n",
    "\n",
    "## Tool Information\n",
    "\n",
    "- **Tool:** AcinetobacterPlasmidTyping (Hamidian et al.)\n",
    "- **Input:** Assembled genome contigs (FASTA)\n",
    "- **Organism:** *Acinetobacter baumannii*\n",
    "- **Analysis type:** Plasmid replicon typing\n",
    "- **Database:** AcinetobacterPlasmidTyping marker database\n",
    "\n",
    "This notebook documents the identification of plasmid replicon types in *Acinetobacter baumannii* genomes using the AcinetobacterPlasmidTyping marker database developed by Hamidian *et al.*\n",
    "\n",
    "The AcinetobacterPlasmidTyping database consists of curated replication initiation (rep) gene sequences representing the major *Acinetobacter* plasmid families, including R1, R3, and RP groups. These markers enable classification of plasmids based on replicon homology.\n",
    "\n",
    "The analysis was performed on assembled genome contigs generated from the GHRU assembly pipeline. Identified plasmid replicon types were used for downstream plasmid epidemiology analysis and integration with antimicrobial resistance and mobile genetic element profiling.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfac328f",
   "metadata": {},
   "source": [
    "## Tool Installation and Cloning\n",
    "\n",
    "The AcinetobacterPlasmidTyping database is publicly available from GitHub:\n",
    "https://github.com/MehradHamidian/AcinetobacterPlasmidTyping\n",
    "\n",
    "The repository contains curated plasmid replication initiation (rep) gene marker sequences and reference documentation for plasmid typing in *Acinetobacter* species.\n",
    "\n",
    "### Cloning the repository\n",
    "\n",
    "The database can be downloaded using `git clone`:\n",
    "`git clone https://github.com/MehradHamidian/AcinetobacterPlasmidTyping.git`\n",
    "\n",
    "Below are the files obtained after cloning, the database itself does not require compilation or installation separately, it is included during cloning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "47046e6e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LICENSE\n",
      "Plasmid_Metadata_20250507.xlsx\n",
      "README.md\n",
      "Reference Plasmids for R1 Reps.pdf\n",
      "Reference Plasmids for R3 reps.pdf\n",
      "Reference Plasmids for RP Reps.pdf\n",
      "marker_db\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/nidhi/tools/AcinetobacterPlasmidTyping\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c4712afa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PROTEIN_acinetobacterplasmidtype_feb2025_V3.fasta\n",
      "acinetobacterplasmidtype_aug2023_v2.fasta\n",
      "acinetobacterplasmidtype_feb2025_V3.fasta\n",
      "acinetobacterplasmidtyper.fasta\n"
     ]
    }
   ],
   "source": [
    "ls /home/nidhi/tools/AcinetobacterPlasmidTyping/marker_db"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1e4f182",
   "metadata": {},
   "source": [
    "## Input Files\n",
    "\n",
    "The input for Acinetobacter Plasmid Typing consists of assembled genome contigs generated from the GHRU assembly pipeline."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0668ad33",
   "metadata": {},
   "source": [
    "## Execution\n",
    "\n",
    "Plasmid replicon typing was performed using BLASTn by screening genome assemblies against the AcinetobacterPlasmidTyping marker database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5d42061c",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /home/nidhi/tools/AcinetobacterPlasmidTyping\n",
    "\n",
    "mkdir -p /data/internship_data/nidhi/aba/new_output/apt_output\n",
    "\n",
    "for f in /data/internship_data/nidhi/aba/new_output/nextflow_output/assemblies/*.fasta; do\n",
    "    sample=$(basename $f .fasta)\n",
    "\n",
    "    blastn \\\n",
    "      -query marker_db/acinetobacterplasmidtyper.fasta \\\n",
    "      -subject $f \\\n",
    "      -outfmt 6 \\\n",
    "      -perc_identity 95 \\\n",
    "      -out /data/internship_data/nidhi/aba/new_output/apt_output/${sample}.tsv\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad323689",
   "metadata": {},
   "source": [
    "## Output Files\n",
    "\n",
    "The output of the Acinetobacter Plasmid Typing analysis consists of per-isolate plasmid replicon typing results generated in tab-separated format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9cc5b429",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ABA-1000.short.tsv\n",
      "ABA-1001.short.tsv\n",
      "ABA-1002.short.tsv\n",
      "ABA-1003.short.tsv\n",
      "ABA-1004.short.tsv\n",
      "ABA-1005.short.tsv\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /data/internship_data/nidhi/aba/new_output/apt_output  | head -n 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "14a437a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2__R3__r3-T5_CP021348.1_pB8300__35      contig00078  99.892   924  1  0  1    924  7664  8587  0.0  1701\n",
      "2__R3__r3-T5_CP038260.1_pEH_gr3__36     contig00078  99.675   924  3  0  1    924  7664  8587  0.0  1690\n",
      "2__R3__r3-T5_CP065434.1_pAB17961-2__37  contig00078  99.783   921  2  0  1    921  7664  8584  0.0  1690\n",
      "2__R3__r3-T5_CP007713.1_pABLAC1__38     contig00078  100.000  924  0  0  1    924  7664  8587  0.0  1707\n",
      "2__R3__r3-T60_CM009647.1_p2ZQ2__125     contig00056  99.735   755  2  0  176  930  128   882   0.0  1384\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "column -t /data/internship_data/nidhi/aba/new_output/apt_output/ABA-1000.short.tsv | head -n 5"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "aba_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
